src/shared/components/Navigation/Navigation.tsx

```diff
@@ -1,58 +1,183 @@
-		setDropdownsOpen(openStates);
-	}, []);
+/**
+ * Main navigation bar component
+ * @param history
+ * @param loginState
+ * @constructor
+ */
+const Navigation: FunctionComponent<NavigationProps> = ({ history, loginState }) => {
+	const [areDropdownsOpen, setDropdownsOpen] = useState<{ [key: string]: boolean }>({});
+	const [isMobileMenuOpen, setMobileMenuOpen] = useState(false);
+
+	const getPrimaryNavigationItems = (): NavigationItem[] => {
+		const isLoggedIn = loginState && loginState.message === LoginMessage.LOGGED_IN;
+		if (isLoggedIn) {
+			return [
+				{ label: 'Home', location: APP_PATH.LOGGED_IN_HOME, key: 'teachers' },
+				{
+					label: 'Zoeken',
+					location: APP_PATH.SEARCH,
+					icon: 'search',
+					key: 'pupils',
+				},
+				{ label: 'Ontdek', location: APP_PATH.DISCOVER, key: 'home' },
+				{
+					label: 'Mijn Werkruimte',
+					location: APP_PATH.WORKSPACE,
+					icon: 'briefcase',
+					key: 'search',
+				},
+				{ label: 'Projecten', location: APP_PATH.PROJECTS, key: 'discover' },
+				{ label: 'Nieuws', location: APP_PATH.NEWS, key: 'workspace' },
+			];
+		} else {
+			return [
+				{ label: 'Voor leerkrachten', location: APP_PATH.FOR_TEACHERS, key: 'teachers' },
+				{ label: 'Voor leerlingen', location: APP_PATH.FOR_PUPILS, key: 'pupils' },
+				{ label: 'Projecten', location: APP_PATH.PROJECTS, key: 'projects' },
+				{ label: 'Nieuws', location: APP_PATH.NEWS, key: 'news' },
+			];
+		}
+	};
+
+	const getSecondaryNavigationItems = (): NavigationItem[] => {
+		const isLoggedIn = loginState && loginState.message === LoginMessage.LOGGED_IN;
+		if (isLoggedIn) {
+			if (isMobileMenuOpen) {
+				return [
+					{ label: 'Instellingen', location: APP_PATH.SETTINGS, key: 'settings' },
+					{ label: 'Hulp', location: APP_PATH.HELP, key: 'help' },
+					{
+						label: 'Rapporteer feedback of probleem',
+						location: APP_PATH.FEEDBACK,
+						key: 'feedback',
+					},
+					{ label: 'Logout', location: APP_PATH.LOGOUT, key: 'logout' },
+				];
+			} else {
+				return [
+					{
+						label: (
+							<div className="c-navbar-profile-dropdown-button">
+								<Avatar initials={getProfileInitials()} name={getProfileName()} />
+								<Icon name="caret-down" size="small" />
+							</div>
+						),
+						component: (
+							<MenuContent
+								menuItems={[
+									[
+										{ id: 'settings', label: 'Instellingen' },
+										{ id: 'help', label: 'Hulp' },
+										{ id: 'feedback', label: 'Rapporteer feedback of probleem' },
+									],
+									[{ id: 'logout', label: 'Logout' }],
+								]}
+								onClick={handleMenuClick}
+							/>
+						),
+						key: 'profile',
+					},
+				];
+			}
+		} else {
+			return [
+				{ label: 'Account aanmaken', component: <PupilOrTeacherDropdown />, key: 'createAccount' },
+				{ label: 'Aanmelden', location: APP_PATH.REGISTER_OR_LOGIN, key: 'login' },
+			];
+		}
+	};
+
+	// useEffect(() => {
```

Comment:
Is dit tijdelijk? Of mag dit volledig weg? Indien tijdelijk, misschien een TODO?